src/app/features/tasks/task/task.component.ts

```diff
@@ -218,6 +221,15 @@ export class TaskComponent implements OnInit, OnDestroy, AfterViewInit {
       setTimeout(() => {
```

Comment:
I'd rather see this happening outside the task component as there can be so many instances of it on the page.

Reply:
I totally agree on this one. In fact I started to look for a good place and wanted to ask but forgot. I think this could be handled once for both archived and active tasks. I couldn't quickly find a common "ancestor" component apart from the app.component and I'm not sure if that's a good place to put it.  What do you think?
Also, could it be that the worklog-page.component isn't used at all? There is no page for it and neither do I see it's selector to appear anywhere.

Reply:
> I think the dataset might not be the most recent version, so this could be the reason why.
> This is what the migrations are for. They are executed whenever a full data set is loaded which can happen on syncing, on app start and on import.

Ok, so I suppose we keep the migration I added.

Reply:
I was hoping to find an even better place to put the route interception. Somewhere it would work for both worklog and task component. work-view is not a parent of worklog afaik.
How would you feel about a NavigationService that intercepts and interacts with DOM by calling .focus(). I know it seems bad practice or unconventional but I think it might work well. It could also make checks like if we are already on the correct page. Which was something that we have both noticed doesn't work perfectly yet. I think this would remove quite a bit of navigation logic from the components. The search bar would then interact exclusively with this service instead of the router itself... Just an idea, haven't thought it through perfectly yet but some parts have worked fairly well in my quick tests. Let me know what you think :)

Reply:
This could also work. Wouldn't  call it NavigationService but rather something more specific, as it likely will be a single purpose service. I would also be completely fine with putting this into the app.component btw. There is room to spare and if we can always move it later on, if we find a better place.

Reply:
So I played around with it and it would definitely work well and allows for a lot of navigation possibilities. Still figured to leave it the way it is now, as it seems quite complete and a "NavigationService" would lead to a complete rewrite. The service could pretty much remove the need of route params for the purpose of searching/finding an item. It would always be there and could interact directly with components instead of relying on params being intercepted by components and handled by the themselves. Just some thoughts...